test/unit/test_mail.py

```diff
@@ -0,0 +1,148 @@
+            'sender@example.com',
+            'recipient@example.com',
+            mock.Mock())
+
+        mock_server.sendmail.assert_called_once_with(
+            'sender@example.com', 'recipient@example.com', mock.ANY)
+
+
+def test_attach_files(mail_notifier):
+    with tempfile.TemporaryDirectory() as temp_dir:
+        file1 = Path(temp_dir) / "file1_run.log"
+        file2 = Path(temp_dir) / "file2_run.log"
+        file1.write_text("file data 1")
+        file2.write_text("file data 2")
+
+        assert file1.exists()
+        assert file2.exists()
+
+        mock_files = [file1, file2]
+
+        with mock.patch("builtins.open", mock.mock_open(read_data="file data")):
+            message = mock.Mock()
+            mail_notifier._attach_files(message, mock_files)
+
+            assert message.attach.call_count == len(mock_files)
+
+
+@pytest.mark.parametrize(
+    "mock_glob_side_effect, send_mail_side_effect, expected_log_message, expected_call_count",
+    [
+        # Normal case: No errors, should not log anything
+        (None, None, None, 0),  # No logs are expected, everything works fine
+
+        # Log connection error: Simulate an error while sending email
+        (None, Exception("SMTP server error"),
+         'An error has occurred while sending a mail for warn about remote_platform', 1),
+
+        # Log attachment error: Simulate failure in file listing (glob raises
+        # exception)
+        (Exception("Failed to list files"), None,
+         'An error has occurred while attaching log files to a warning email about remote_platforms ',
+         1)
+    ],
+    ids=[
+        "Normal case: No errors",
+        "Log connection error (SMTP server error)",
+        "Log attachment error (Failed to list files)"
+    ]
+)
+def test_notify_experiment_status(
+        mail_notifier,
+        mock_glob_side_effect,
+        send_mail_side_effect,
+        expected_log_message,
+        expected_call_count):
+    original_local_root_dir = BasicConfig.LOCAL_ROOT_DIR
+    with tempfile.TemporaryDirectory() as temp_dir:
+        BasicConfig.LOCAL_ROOT_DIR = temp_dir
+        exp_id = '1234'
+        aslogs_dir = Path(temp_dir) / exp_id / "tmp" / "ASLOGS"
+        aslogs_dir.mkdir(parents=True, exist_ok=True)
+
+        file1 = aslogs_dir / "1234_run.log"
+        file2 = aslogs_dir / "1235_run.log"
+
+        file1.write_text("file data 1")
+        file2.write_text("file data 2")
+
+        assert file1.exists()
+        assert file2.exists()
+
+        with mock.patch.object(mail_notifier, '_generate_message_experiment_status',
+                               return_value="Test message"), \
+                mock.patch.object(mail_notifier, '_send_mail') as mock_send_mail, \
+                mock.patch.object(mail_notifier, '_attach_files') as mock_attach_files, \
+                mock.patch.object(Log, 'printlog') as mock_printlog:
+
+            mail_to = ['recipient@example.com']
+            platform = mock.Mock()
```

Comment:
We have `Platform` imported at the top, but not used. If you'd like you can add it here,

```
platform: Platform = mock.Mock()
```

this will instruct mypy/IDE/etc that `platform` is a `platform`... and then auto-complete should work (and other type checking tools...). Or remove the type checking import. It's shown as not used and adds one warning to the file.

![image](https://github.com/user-attachments/assets/53929aa2-9463-4dad-a6da-5d1a98bbf1b0)